test/integ/test_timestamp_binding.py

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from datetime import datetime
+from typing import List
+
+import pytest
+
+from snowflake.connector.cursor import SnowflakeCursor
+
+from ..randomize import random_string
+
+
+def create_or_replace_table(cur: SnowflakeCursor, table_name: str, columns: List[str]):
+    sql = f"CREATE OR REPLACE TEMP TABLE {table_name} ({','.join(columns)})"
+    cur.execute(sql)
+
+
+def insert_multiple_records(
+    cur: SnowflakeCursor,
+    table_name: str,
+    ts: str,
+    row_count: int,
+    should_bind: bool,
+):
+    sql = f"INSERT INTO {table_name} values (?)"
+    dates = [[ts] for _ in range(row_count)]
+    cur.executemany(sql, dates)
+    is_bind_sql_scoped = "SHOW stages like 'SNOWPARK_TEMP_STAGE_BIND'"
+    is_bind_sql_non_scoped = "SHOW stages like 'SYSTEMBIND'"
+    res1 = cur.execute(is_bind_sql_scoped).fetchall()
+    res2 = cur.execute(is_bind_sql_non_scoped).fetchall()
+    if should_bind:
+        assert len(res1) != 0 or len(res2) != 0
+    else:
+        assert len(res1) == 0 and len(res2) == 0
+
+
+@pytest.mark.parametrize(
+    "timestamp_type, timestamp_precision",
+    [
+        ("TIMESTAMPTZ", 6),
```

Comment:
do we need to test other timestamp types:
- timestamp_ltz
- timestamp_ntz
- timestamp

and other precision?